### ISSUE TITLE
Bug: [no-unsafe-return] Differentiate a types-error any from a "true" any

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9218]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&fileType=.ts&code=DYUwLgBAbghsCuIBcEByB7MBpAdug7jgNwBQJAZvDgMZgCW6OEIAHjALYAOoAFAJQQA3iQgQATuHhimsBCFIBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolrRE%2BWNCbooiaNA7RI4MAF8QKoA&tsconfig=&tokens=false

### Repro Code

```TypeScript
let value: NotKnown;

function example() {
  return value;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unsafe-return": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

When the value being returned is of an "error" type, there should be some indication that it's not the same as an `any`. It's confusing for developers to see both "error" `any`s and "true" `any`s described with the same messaging.

### Actual Result

```plaintext
Unsafe return of an `any` typed value. 4:3 - 4:16
```

### Additional Info

Split out of #8304 <- #8231.

💖
